spam/spam_classification.sh

```diff
@@ -0,0 +1,322 @@
+        words[$((i))]=$line
+        firstletter="$( echo $line | head -c 1 )"
+        if [ "$firstletter" != "$letter" ]
+        then
+                letterstartind[$((lettercount))]=$((i))
+                letterstart[$((lettercount))]=$firstletter
+                letter=$firstletter
+                let "lettercount++"
+        fi
+        let "i++"
+done < words.txt
+letterstartind[$((lettercount))]=$((i))
+echo "Created list of letters"
+
+touch wordfrequency.txt
+rm wordfrequency.txt
+touch wordfrequency.txt
+messagecount=0
+messagenum=0
+messages="$( wc -l messages.txt )"
+i=0
+while IFS= read -r line; do
+        let "messagenum++"
+        declare -a wordcount=()
+        declare -a wordarray=()
+        read -r -a wordarray <<< "$line"
+        let "messagecount++"
+        words=${#wordarray[@]}
+        for word in "${wordarray[@]}"; do
+                startletter="$( echo $word | head -c 1 )"
+                j=-1
+                while [ $((j)) -lt $((lettercount)) ]; do
+                        let "j++"
+                        if [ "$startletter" == "${letterstart[$((j))]}" ]
+                        then
+                                mystart=$((j))
+                        fi
+                done
+                myend=$((mystart))+1
+                j=${letterstartind[$((mystart))]}
+                jend=${letterstartind[$((myend))]}
+                while [ $((j)) -le $((jend)) ]; do
+                        wordcount[$((j))]=0
+                        if [ "$word" == "${words[$((j))]}" ]
+                        then
+                                wordcount[$((j))]="$( echo $line | grep -o $word | wc -l )"
+                        fi
+                        let "j++"
+                done
+        done
+        for j in "${!wordcount[@]}"; do
+                wordcount[$((j))]=$(echo " scale=4; $((${wordcount[$((j))]})) / $((words))" | bc)
+        done
+        wordcount[$((words))+1]=$((words))
+        echo "${wordcount[*]}" >> wordfrequency.txt
+        echo "Processed message ""$messagenum"
+        let "i++"
+done < messagesLetters.txt
+
+# Create csv file
+tr ' '  ',' < wordfrequency.txt > data.csv
+
+<<COMMENT
+Since Bash is an interpreted language, this simple implementation can 
+take up to 30 minutes to complete.
+COMMENT
+
+<<COMMENT
+Once the script has finished running, split the data into testing (30%) 
+and training (70%) sets:
+COMMENT
+
+mlpack_preprocess_split                        \
+    --input_file data.csv                      \
+    --input_labels_file labels.csv             \
+    --training_file train.data.csv             \
+    --training_labels_file train.labels.csv    \
+    --test_file test.data.csv                  \
+    --test_labels_file test.labels.csv         \
+    --test_ratio 0.3                           \
+    --verbose
+
+<<COMMENT
+Now train a Logistic regression model
+(https://mlpack.org/doc/mlpack-3.3.1/cli_documentation.html#logistic_regression):
```

Comment:
```suggestion
(https://mlpack.org/doc/stable/cli_documentation.html#logistic_regression):
```
It's probably better to just link to the stable documentation. :)